### DISCUSSION TITLE
Reference project root directory?

### DISCUSSION THREAD

**Discussion by Mrodent** [pytest-dev/pytest#8891] [Q&A] [answered]
pytest is wonderful, thanks to all.

I just have a small question: is there any way of identifying, in the context of tests, the project root directory, by which I mean the directory with the src and tests directories? The problem being that you can run `pytest ... ` from anywhere, be it the root directory or anywhere under [root]/tests, and that location then appears to become the CWD.

Incidentally, in order to exclude a directory under my root directory from test discovery searching, I found that I had to put a pytest.ini in that root directory... which then also seemed to alter the location of the CWD (always making it the root directory).  

So it can be a bit of a rigmarole to find where the root directory may actually be in relation to the CWD, and I just wonder whether there's a simple way of obtaining that root directory path.

**Comment by bluetech** [comment, marked as answer]
You should get the `pytestconfig` fixture, then it's `pytestconfig.rootpath` (it's a `pathlib.Path` instance). This returns the [`rootdir`](https://docs.pytest.org/en/latest/reference/customize.html#rootdir).

**Reply by Mrodent** [reply]
Thanks
